### PR TITLE
Implement GSP-109: Redesign Features

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

File: go/testdata/service.toml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+virtual = ["offset"]
+
+[namespace.storage.op.create]
+optional = ["object_mode"]
+
+[namespace.storage.op.stat]
+optional = ["object_mode"]
+
+[namespace.storage.op.write]
+simulated = true
```
Comment (JinnyYi) [inline go/testdata/service.toml:48]: Maybe `simulated` is no longer valid?

**Comment by Xuanwo** [comment]
@JinnyYi PTAL

File: go/spec.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ type Field struct {
 // Service is the data parsed from TOML.
 type Service struct {
 	Name       string
+	Features   []string
```
Comment (xxchan) [inline go/spec.go:53]: Should this be `Features`?

**Review by JinnyYi — APPROVED** [review]

File: go/spec.go (hunk 2 above)
Comment (Xuanwo) [reply go/spec.go:53]: Feature in service should be a string array like:

```toml
features = ["virtual_dir"]
```

And it's not the same type as defined in `features.toml`. We will check them while parsing `service.toml`

Comment (Xuanwo) [reply go/spec.go:53]: I will add comments for these behavior.

Comment (Xuanwo) [reply go/spec.go:53]: Or maybe we need to put them in namespace instead.

Comment (Xuanwo) [reply go/spec.go]: PTAL, I moved features from the service global to namespace level.

**Review by xxchan — APPROVED** [review]

**Pull request merged**